classes/class15/README.md

```diff
@@ -1 +1,72 @@
+
+##Before Agile
+Talking about agile is easier when you are first aware of what was the style before agile. 
+In the past, software would be developed using what has become known as waterfall stragety. In this style, the project is tackled by first doing the requirements for the entire project, the design, then the code, and finally one giant release after which the only work on the project would be maintenance. This made sense when software was released largely on discs or even downloads with no updating. Now, however, we develop websites that can change at any frequency without any extra work beign done on the part of the user. There are lot of other components too, but a big part of agile rapid iteration and releases. Make things quickly and push your code to your users frequently, taking and responding to feedbak as you go. 
+
+##Agile Manifesto
+The agile manifesto is one publication that tries to specify some of the agile principles. 
+They list these four piorities as being central to the agile methodology: 
+
+- Individuals and interactions over Processes and tools
+- Working software over Comprehensive documentation
+- Customer collaboration over Contract negotiation
+- Responding to change over Following a plan
+
+More specifically, there are 12 principles that also help define agile: 
+
+- Customer satisfaction by rapid delivery of useful software
+- Welcome changing requirements, even late in development
+- Working software is delivered frequently (weeks rather than months)
+- Close, daily cooperation between business people and developers
+- Projects are built around motivated individuals, who should be trusted
+- Face-to-face conversation is the best form of communication (co-location)
+- Working software is the principal measure of progress
+- Sustainable development, able to maintain a constant pace
+- Continuous attention to technical excellence and good design
+- Simplicity—the art of maximizing the amount of work not done—is essential
+- Self-organizing teams
+- Regular adaptation to changing circumstance
+
+##Scrum
+As mentioned ealier, agile is more of a mindset and methodology than an actual set of instructions on exactly what to do. A very popular addition to agile that brings in more specific instructions is the scrum framework. In fact, scrum is used so often with agile that components of one are often mistaken for components of the other. What srum adds is a set of roles and events to your development process that are meant to help maximize your efficiency. It is worth noting that while there are "official" scrum methods, the actual use and implementation varies a good amount from company to company and even team to team. 
+
+https://www.youtube.com/watch?v=oyVksFviJVE
+
+###Stories
+A big part of scrum is breaking down the tasks of a project into what are called stories. 
+These stories could be features, bugs, learning about something, and so on. 
```

Comment:
http://agileatlas.org/articles/item/spikes-in-scrum-the-exception-not-the-rule
Basically, a story that has an "answer" or knowledge as a deliverable rather than code. Research the choice of a framework might be considered a spike.